### PR TITLE
Focus management

### PR DESCRIPTION
**Pull request by WhoAteDaCake**
Very basic implementation of focus management.

Still requires testing.

### CODE REVIEW

**Comment by WhoAteDaCake** [comment]
Not sure if this the ideal approach though. Some code seems a bit messy.
Also might be a good time to update the way props are passed to views

File: src/UI/Focus.re [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+type handleEvent = event => unit
+type active = {
+	handler: handleEvent,
+	id: int
+}
+type focused = ref(option(active));
+let focused = ref(None);
+
+/* Should happen when user clicks anywhere where no focusable node exists */
+let looseFocus = () => switch (focused^) {
```
Comment (bryphe) [inline src/UI/Focus.re]: Nit: `looseFocus` has an extra `o`

File: src/UI/Node.re [hunk 2]
```diff
@@ -44,6 +46,8 @@ class node ('a) (()) = {
   };
   pub getInternalId = () => _internalId;
   pub measurements = () => _layoutNode^.layout;
+  pub getTabIndex = () => _tabIndex^;
```
Comment (bryphe) [inline src/UI/Node.re]: Thanks for adding the `tabIndex` 👍

**Comment by WhoAteDaCake** [comment]
Thanks for feedback @bryphe I've updated the code and added few tests.

**Comment by WhoAteDaCake** [comment]
I think I just forgot to change it. If you are happy with this, it can be merged

**Comment by bryphe** [comment]
Sounds good @WhoAteDaCake ! Merging now - thanks for your work here!

**Pull request merged**